bank.py

```diff
@@ -77,16 +79,19 @@ def show_balance(self, user: User) -> None:
         print(f"You choosed the Show balance operation!\n"
               f"Your current balance is {user.balance}")
 
-    def save_users(self):
-        """loads all information about users"""
-        with open("database.txt", "w") as f:
-            for user in self.users:
-                f.write(f"{user.user_id}:{user.pin}:{user.balance}\n")
-
     def change_user_pin(self, user: User, new_pin: str) -> None:
         """changes the user's pin"""
         if not new_pin:
             return
 
         user.pin = new_pin
         self.save_users()
+
+    def save_users(self) -> None:
+        """loads all information about users in JSON"""
+        for user in self.users:
+            user_json = {"login": user.user_id, "pass": user.pin, "balance": user.balance}
```

Comment:
you don't need to create dict by yourself.

Please try to use JSON.dumps which returns str, and put this string to file.

link to read:
https://stackoverflow.com/questions/26033239/list-of-objects-to-json-with-python

Reply:
btw: structure of file may changed it is okay